indsl/ts_utils/array_operations.py

```diff
@@ -0,0 +1,127 @@
+import numpy as np
+
+# Simple operations
+import pandas as pd
+
+from indsl.exceptions import UserValueError
+from indsl.type_check import check_types
+
+
+@check_types
+def time_weighted_mean(data: pd.Series) -> pd.Series:
+    """Time weighted mean.
+
+    The time weighted mean for a time series. The returned timeseries has a constant value
+
+    Args:
+        data: Time series.
+
+    Returns:
+        pandas.Series: Time weighted mean.
+    """
+    from scipy.integrate import trapezoid
+
+    n = len(data)
```

Comment:
you can use `validate_series_is_not_empty` function that is implemented in `validations.py` so you will avoid repeating the code